v-next/hardhat/test/internal/builtin-plugins/network-manager/type-validation.ts

```diff
@@ -0,0 +1,396 @@
+import type {
```

Comment:
In these tests the errors are checked in the following way:
```
  assert.notEqual(validationErrors.length, 0);
  assert.equal(validationErrors[0].message, errMsg);
```

and not like this:
```
      assert.deepEqual(
        validateNetworkConfig(config),
        [
          {
            message: "errMsg",
            path: ["paths"],
          },
        ],
      );
```
Because the error return usually is quite verbose, e.g.:
```
  {
    code: 'invalid_union',
    unionErrors: [ [ZodError], [ZodError], [ZodError] ],
    path: [ 'accounts' ],
    message: "Expected 'remote', an array of strings, or an object with account details"
  }
```